packages/docusaurus-theme-classic/src/types.d.ts

```diff
@@ -258,9 +258,43 @@ declare module '@theme/Layout' {
   export default Layout;
 }
 
+declare module '@theme/LayoutHead' {
+  import type {Props} from '@theme/Layout';
+
+  const LayoutHead: (props: Props) => JSX.Element;
+  export default LayoutHead;
+}
+
+declare module '@theme/SearchMetadatas' {
+  export type Props = {
+    locale?: string;
+    version?: string;
+    tag?: string;
+  };
+
+  const SearchMetadatas: (props: Props) => JSX.Element;
+  export default SearchMetadatas;
+}
+
+declare module '@theme/LastUpdated' {
+  export type Props = {
```

Comment:
i'm unsure if using `Props` as a type everywhere is a good idea, but in 90% of cases thats how its called in project, 

i was tempted to rename all of them to `{ComponentName}Props` but this will be a bigger change and if that is desirable i can update it

Reply:
don't know

Honestly not very satisfied with our current TS setup, this is a bit annoying to maintain and I'd rather emit types based on our actual code rather than hand-write it as we do today.

Reply:
ok i'm going to prepare PR for that :)